### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by kepengxu** [RenYang-home/RLVC#2]
Thank you very much for your project code. Do you have plans to open source the code for the RLVC training phase.

**Comment by RenYang-home** [comment]
We currently won't release the training code, but the training process is similar to the training code of OpenDVC (https://github.com/RenYang-home/OpenDVC), just plus the last phase to train the total loss of the whole network on sequential frames.

**Comment by cyhhhhhhit** [comment]
hi， i have a problem while training RPM with ConvLSTMCell. 

I can ignore it by disable layout optimizer, but i am confused whether it works?

>`layout failed: Invalid argument: MutableGraphView::SortTopologically error: detected edge(s) creating cycle(s) {'MV_analysis/recurrent/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'MV_analysis/recurrent/rnn/while/rnn/conv_lstm_cell/concat', 'analysis/recurrent/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'analysis/recurrent/rnn/while/rnn/conv_lstm_cell/concat', 'synthesis/recurrent/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'synthesis/recurrent/rnn/while/rnn/conv_lstm_cell/concat', 'RNNRES/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'RNNRES/rnn/while/rnn/conv_lstm_cell/concat', 'MV_synthesis/recurrent/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'MV_synthesis/recurrent/rnn/while/rnn/conv_lstm_cell/concat', 'RNNMV/rnn/while/rnn/conv_lstm_cell/concat-1-TransposeNHWCToNCHW-LayoutOptimizer' -> 'RNNMV/rnn/while/rnn/conv_lstm_cell/concat'}.`

**Comment by RenYang-home** [comment]
Do you have this problem when training RAE?

**Comment by cyhhhhhhit** [comment]
yes, i found it related this [issue](https://github.com/keras-team/tf-keras/issues/530)

**Comment by RenYang-home** [comment]
Thanks for sharing the issue. Have you sloved the problem now?

**Comment by cyhhhhhhit** [comment]
I basically hide this error by
`tf.config.optimizer.set_experimental_options({'layout_optimizer': False})`.
It seems wont degrade reconstruction performance.

**Comment by RenYang-home** [comment]
Okay, I never met this before. Good to know more about it.
